config/config/src/main/java/io/helidon/config/UrlConfigSource.java

```diff
@@ -162,25 +160,8 @@ private String mediaType(String responseMediaType) throws URISyntaxException {
         return mediaType;
     }
 
-    private String probeContentType() throws URISyntaxException {
-        URI uri = url.toURI();
-        Path path;
-        switch (uri.getScheme()) {
-            case "jar":
-                String relativePath = uri.getSchemeSpecificPart();
-                int idx = relativePath.indexOf("!");
-                if (idx > 0 && idx < relativePath.length()) {
-                    relativePath = relativePath.substring(idx + 1);
-                }
-                path = Paths.get(relativePath);
-                break;
-            case "file":
-                path = Paths.get(uri);
-                break;
-            default:
-                path = Paths.get(url.getPath());
-        }
-        return ConfigHelper.detectContentType(path);
+    private String probeContentType() {
+        return MediaTypes.detectType(url).orElse(null);
```

Comment:
Very minor point. The `probeContentType` methods in the other classes return `Optional<String>`. This one returns `String` and uses `orElse`. After a quick look I did not see any null checks on the returned `String` from this method. Just for consistency would it make sense (without excessive disruption of the other sites which invoke `UrlConfigSource#probeContentType`) to  return the `Optional` just as `probeContentType` in the other classes do?

Reply:
I wanted to simply replace the existing usage. In other classes, `Optional` was removed. This class just returned the value (which could have been `null` with previous implementation).
I will have a look if this is not a too complicated change to do

Reply:
This seems to be a deeper problem in the implementation, as `ConfigParser.Content.mediaType()` is allowed to return null. I will create a follow up issue to clean this up

Reply:
Follow up: https://github.com/oracle/helidon/issues/1333